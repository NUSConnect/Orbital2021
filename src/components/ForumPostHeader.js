import React from "react";
import { StyleSheet, Text, TouchableOpacity, View } from "react-native";
import { Ionicons } from "react-native-vector-icons";

export default function SubForumHeader({ title, style, goBack, ...props }) {
    return (
        <View style={styles.container}>
            <TouchableOpacity
                style={styles.button}
                activeOpacity={0.4}
                onPress={goBack}
                testID="back"
            >
                <Ionicons
                    name="arrow-back"
                    color="#79D2E6"
                    size={38}
                    style={styles.icon}
                />
            </TouchableOpacity>
            <Text style={styles.text}>{title}</Text>
            <Ionicons
                name="add"
                color="white"
                size={38}
                style={{ paddingLeft: 12 }}
            />
        </View>
    );
}

const styles = StyleSheet.create({
    container: {
        height: 60,
        flexDirection: "row",
        backgroundColor: "#ffffff",
        borderColor: "#dcdcdc",
        borderWidth: 1,
        alignItems: "center",
        justifyContent: "center",
    },
    button: {
        width: "15%",
        paddingLeft: 12,
    },
    text: {
        flex: 1,
        width: "85%",
        color: "#ff7f50",
        fontSize: 30,
        textAlign: "center",
    },
    icon: {},
});