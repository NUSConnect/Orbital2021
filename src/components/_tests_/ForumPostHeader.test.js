import * as React from "react";
import ForumPostHeader from "../ForumPostHeader";
import renderer from "react-test-renderer";
import { fireEvent, render } from '@testing-library/react-native';

it(`renders correctly`, () => {
    const tree = renderer.create(<ForumPostHeader />);
    expect(tree).toMatchSnapshot();
});

it(`on press button`, () => {
    const mockGoBackFn = jest.fn();
    const {getByTestId} = render(<ForumPostHeader goBack={mockGoBackFn} />);
    fireEvent.press(getByTestId('back'))
    expect(mockGoBackFn).toHaveBeenCalled();
});