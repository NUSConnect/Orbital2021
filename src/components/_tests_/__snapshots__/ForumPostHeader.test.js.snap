// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#ffffff",
      "borderColor": "#dcdcdc",
      "borderWidth": 1,
      "flexDirection": "row",
      "height": 60,
      "justifyContent": "center",
    }
  }
>
  <View
    accessible={true}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
        "paddingLeft": 12,
        "width": "15%",
      }
    }
    testID="back"
  />
  <Text
    style={
      Object {
        "color": "#ff7f50",
        "flex": 1,
        "fontSize": 30,
        "textAlign": "center",
        "width": "85%",
      }
    }
  />
</View>
`;
